src/lib/components/Footer/FooterCopyright.tsx

```diff
@@ -0,0 +1,21 @@
+import classNames from 'classnames';
+import { FC, PropsWithChildren } from 'react';
+
+export type CopyrightProps = PropsWithChildren<{
+  href?: string;
+  by?: string | undefined;
+  year?: number;
+  className?: string;
+}>;
+
+export const FooterCopyright: FC<CopyrightProps> = ({ href, by, year, className }) => {
+  return (
+    <span className={classNames('text-sm  text-gray-500 dark:text-gray-400 sm:text-center', className)}>
+      © {year}
+      <a href={href} className="ml-1 hover:underline">
```

Comment:
here, you need to control the flow, since the `href` and `by` are optional. I guess that, `by` should be required. Probably you want to just check if `href` is set, then render it as a link, otherwise render it just as a span.

Reply:
You need to handle optional values at the template flow, or you need to make them required.